src/Http/RequestFactory.php

```diff
@@ -60,25 +60,55 @@ public function setProxy($proxy)
 	 */
 	public function createHttpRequest()
 	{
+		$remoteAddr = !empty($_SERVER['REMOTE_ADDR']) ? $_SERVER['REMOTE_ADDR'] : NULL;
+		$remoteHost = !empty($_SERVER['REMOTE_HOST']) ? $_SERVER['REMOTE_HOST'] : NULL;
+
+		// use real client address and host if trusted proxy is used
+		$usingTrustedProxy = (bool)count(array_filter($this->proxies, function ($proxy) use ($remoteAddr) {
```

Comment:
Sorry, count was a leftover, originally I had it like `count(...) === 0;` but then I changed my mind. :)